Source/WebCore/workers/service/server/SWServer.cpp

```diff
@@ -806,9 +806,8 @@ LastNavigationWasAppInitiated SWServer::clientIsAppInitiatedForRegistrableDomain
     return LastNavigationWasAppInitiated::No;
 }
 
-void SWServer::tryInstallContextData(ServiceWorkerContextData&& data)
+void SWServer::tryInstallContextData(const RegistrableDomain& registrableDomain, ServiceWorkerContextData&& data)
 {
-    RegistrableDomain registrableDomain(data.scriptURL);
```

Comment:
I think we should keep data.scriptURL for that registrable domain.
If a service worker is serving an iframe of origin A embedded in a top level page of origin B, we want to select the process with origin A, not origin B.

But we still need to let know that originB is a now a valid registrable domain for the process running the service worker. This part seems missing